File: backend/app/schemas/workspace_collaboration.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class WorkspaceCollaborationCreate(BaseModel):
    item_type: str
    title: str | None = None
    body: str | None = None
    status: str | None = None
    priority: str | None = None
    mentions: list[str] = Field(default_factory=list)
    assignee_user_id: str | None = None
    due_date: date | None = None


class WorkspaceCollaborationUpdate(BaseModel):
    title: str | None = None
    body: str | None = None
    status: str | None = None
    priority: str | None = None
    mentions: list[str] | None = None
    assignee_user_id: str | None = None
    due_date: date | None = None


class WorkspaceCollaborationResponse(BaseModel):
    collaboration_id: str
    workspace_id: str
    item_type: str
    title: str | None = None
    body: str | None = None
    status: str | None = None
    priority: str | None = None
    mentions: list[str] = Field(default_factory=list)
    assignee_user_id: str | None = None
    created_by_user_id: str | None = None
    due_date: date | None = None
    resolved_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class WorkspaceCollaborationListResponse(BaseModel):
    items: list[WorkspaceCollaborationResponse] = Field(default_factory=list)
    summary: dict[str, int] = Field(default_factory=dict)

File: frontend/src/types/workspaceCollaboration.ts
export type WorkspaceCollaborationItemType = 'comment' | 'task' | 'review' | string;

export interface WorkspaceCollaborationItem {
  collaboration_id: string;
  workspace_id: string;
  item_type: WorkspaceCollaborationItemType;
  title?: string | null;
  body?: string | null;
  status?: string | null;
  priority?: string | null;
  mentions: string[];
  assignee_user_id?: string | null;
  created_by_user_id?: string | null;
  due_date?: string | null;
  resolved_at?: string | null;
  created_at: string;
  updated_at: string;
}

export interface WorkspaceCollaborationSummary {
  total_items: number;
  open_items: number;
  completed_items: number;
  comment_count: number;
  task_count: number;
  review_count: number;
  mention_count: number;
}

export interface WorkspaceCollaborationListResponse {
  items: WorkspaceCollaborationItem[];
  summary: WorkspaceCollaborationSummary;
}

export interface WorkspaceCollaborationCreatePayload {
  item_type: WorkspaceCollaborationItemType;
  title?: string | null;
  body?: string | null;
  status?: string | null;
  priority?: string | null;
  mentions?: string[];
  assignee_user_id?: string | null;
  due_date?: string | null;
}

export interface WorkspaceCollaborationUpdatePayload {
  title?: string | null;
  body?: string | null;
  status?: string | null;
  priority?: string | null;
  mentions?: string[];
  assignee_user_id?: string | null;
  due_date?: string | null;
}
